### PR TITLE
nintendo/n64_gateway.cpp: New driver and software list submission [Foxhack, Angelo Salese]

### PR DESCRIPTION
**Pull request by FoxhackDN**
New software list items marked not working
------------------------------------------
n64_lodgenet.xml: 1080° Snowboarding, Donkey Kong 64, Dr. Mario, Excitebike 64, F-Zero X, Gauntlet Legends, Hydro Thunder, Kirby 64 - The Crystal Shards, The Legend of Zelda - Majora's Mask, The Legend of Zelda - Ocarina of Time, Mario Golf, Mario Kart 64, Mario Party 3, Mario Tennis, Midway's Greatest Arcade Hits Vol. 1, Mortal Kombat 4, The New Tetris, Paper Mario, Pilotwings 64, Pokémon Snap, Rampage 2 - Universal Tour, Ready 2 Rumble Boxing, Rush 2 - Extreme Racing USA, Star Fox 64, Star Wars - Rogue Squadron, Super Mario 64, Super Smash Bros., Virtual Chess, Virtual Pool 64, Wave Race 64, Yoshi's Story [Forest of Illusion]

### CODE REVIEW

**Comment by FoxhackDN** [comment]
This is a draft software list submission for @angelosa 's planned LodgeNet 64 additions, which he started to work on here: https://github.com/mamedev/mame/commit/ba698741ddd2ef885403c47a0bf3b81130cc558a . It won't be used until that system is finished, but I figured I should submit this for easier system testing, and if things need to be changed on my end (archive names, XML fixes, documentation, etc.)

Every game includes a z64 dump and an i64 file that contains configuration info used by the LodgeNet 64 system. The i64 file hashes are referenced but the files are not included in the game archives (I didn't know how to add them as extra unused files.) These files were released in decrypted format, so they're not proper dumps and I made a note that they should be considered bad dumps for the moment.

Comments are appreciated. I'm sure I messed up a lot of things.

File: hash/n64_lodgenet.xml [hunk 1]
```diff
@@ -0,0 +1,479 @@
...
+	</software>
+
+	<software name="marioten">
+		<description>Mario Tennis</description>
+		<year>2000</year>
+		<publisher>Nintendo</publisher>
+		<info name="serial" value="nm8g00"/>
+		<part name="cart" interface="n64_cart">
+			<dataarea name="rom" size="0x1000000">
+				<rom name="nm8g00.z64" size="0x1000000" crc="02af0a24" sha1="1f76e775c5f2e247d6550aecc2666149c60259
```
Comment (angelosa) [inline hash/n64_lodgenet.xml:234]: This line is mangled, need the full sha1 to fix.

Comment (FoxhackDN) [reply hash/n64_lodgenet.xml]: Fixed now, thank you!

File: src/mame/nintendo/n64_lodgenet.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
+// license:BSD-3-Clause
+// copyright-holders:FoxhackDN, Angelo Salese
+/*
+    Nintendo Gateway 64
+
+    <fill me>
```
Comment (angelosa) [inline src/mame/nintendo/n64_lodgenet.cpp:6]: This needs notes moving from the .xml to here.

File: src/mame/nintendo/n64_lodgenet.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    TODO:
+    - Set-top box part;
+    - Extra joypad buttons;
+    - Doesn't really have cartslot but rather uploads games thru a DOS/V Windows 95/98 program
+      that works with encrypted files.
+      \- Notice that the available dump is "patched to work on emulators",
+         it may normally expect a specific ISA/PCI board setup plus a setup/install CD-ROM;
```
Comment (angelosa) [inline src/mame/nintendo/n64_lodgenet.cpp:14]: Given this and previous @cuavas concerns I'm very inclined to just c&p the cartslot stuff and derive from n64 base state instead (given that it eventually need to be removed if/when the real PC interface is working). 
It also needs those i64 files to actually be defined, something I'm currently not sure about how to do in XML form.

**Comment by angelosa** [comment]
I don't think it's fair to consider these working, mind to make a bulk replace with `supported="no"`?

**Comment by FoxhackDN** [comment]
> * [ ]  I don't think it's fair to consider these working, mind to make a bulk replace with `supported="no"`?

As soon as I can figure out how to do it without nuking my own branch, sure.

**Comment by cuavas** [comment]
The trouble with this is the software here is effectively quickload images.  The system doesn’t use a cartridge slot at all, and video is streamed to remote terminals.  It’s probably going to be very difficult to find enough information and dump enough media to emulate this properly.

**Comment by angelosa** [comment]
The player runs on DOS/V SCSI (PCI?) card, decrypt the original files, sends the eeerase thing to N64 client for factory defaults then streams the actual game. And of course there's the other side of the medal that is the set top box setting. It's almost an hopeless cause.

**Pull request merged**